### PR TITLE
Chore: Added MVX Gas Prices in USD

### PR DESCRIPTION
**Pull request by muktanshumishra24**
# Pull Request Template

Resolves or continues [AR-8282](...).

## Changes
**Added MVX Gas Prices in USD**

Screenshot 2024-07-16 at 03 39 14
Screenshot 2024-07-16 at 03 39 08
Screenshot 2024-07-17 at 03 40 43
Screenshot 2024-07-17 at 03 41 03

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

[AR-8282]: https://team-1624093970686.atlassian.net/browse/AR-8282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/CustomRequestScreen/Mvx/MVXSignTransaction.vue [hunk 1]
```diff
@@ -25,5 +25,14 @@ const props = defineProps<{ transaction: any }>()
         }}
       </span>
     </div>
+    <div class="flex justify-between gap-4 text-base">
+      <span class="w-[120px] capitalize">Gas Fees</span>
+      <span
+        >{{
+          transaction.gasLimit * transaction.gasPrice * Math.pow(10, -18)
```
Comment (mmjee) [inline src/components/CustomRequestScreen/Mvx/MVXSignTransaction.vue]: Do not use ordinary `Number`s for calculations on this scale, it'll overflow immediately, use `Decimal`s.

File: src/pages/SendNft.vue [hunk 2]
```diff
@@ -320,20 +320,26 @@ async function determineGasParamsMVX(gasLimitInput: string | number = 0) {
...
+  const gasLimitDecimal = new Decimal(
+    Number(gasLimitInput) || minGasLimitDecimal
+  )
+  gasParamsMVX.value.gasLimit = Number(gasLimitDecimal)
 
-  gasParamsMVX.value.gasFee = parseFloat(gasFee.toFixed(5))
+  const gasFeeDecimal = gasLimitDecimal
+    .times(gasPriceDecimal)
+    .div(new Decimal(currencyStore.currencies['EGLD']))
+  gasParamsMVX.value.gasFee = Number(new Decimal(gasFeeDecimal.toFixed(5)))
```
Comment (mmjee) [inline src/pages/SendNft.vue]: Why do this? Could just be `gasFeeDecimal.toFixed(5).toNumber()`.

File: src/pages/SendTokens.vue [hunk 3]
```diff
@@ -206,32 +206,37 @@ onUnmounted(() => {
...
 
-  const gasFee =
-    (gasParamsMVX.value.gasLimit * gasParamsMVX.value.gasPrice) /
-    currencyStore.currencies['EGLD']
+  const gasFeeDecimal = gasLimitDecimal
+    .times(gasPriceDecimal)
+    .div(new Decimal(currencyStore.currencies['EGLD']))
 
-  gasParamsMVX.value.gasFee = parseFloat(gasFee.toFixed(5))
+  gasParamsMVX.value.gasFee = parseFloat(
```
Comment (mmjee) [inline src/pages/SendTokens.vue]: See above comment.

File: src/components/CustomRequestScreen/Mvx/MVXSignTransactions.vue [hunk 4]
```diff
@@ -34,12 +54,7 @@ const currencyStore = useCurrencyStore()
           <span class="w-[120px] capitalize">Gas Fees</span>
           <span
             >{{
-              (
-                (transaction.gasLimit *
-                  transaction.gasPrice *
-                  Math.pow(10, -18)) /
-                currencyStore.currencies['EGLD']
-              ).toFixed(5)
+              calculateGasFeeUSD(transaction, currencyStore.currencies['EGLD'])
```
Comment (mmjee) [inline src/components/CustomRequestScreen/Mvx/MVXSignTransactions.vue]: Why not make this a `computed`? What's the purpose of a function if you are not going to re-use it and scope it to this component?

File: src/pages/SendNft.vue (hunk 2 above)
Comment (muktanshumishra24) [reply src/pages/SendNft.vue]: toFixed Returns a String which while using toNumber() throws 
`Property 'toNumber' does not exist on type 'string'.ts(2339)`

Using To Decimal Places now, FIXED.

File: src/pages/SendTokens.vue [hunk 5]
```diff
@@ -234,9 +234,9 @@ async function determineGasParamsMVX(gasLimitInput: string | number = 0) {
     .times(gasPriceDecimal)
     .div(new Decimal(currencyStore.currencies['EGLD']))
 
-  gasParamsMVX.value.gasFee = parseFloat(
-    new Decimal(gasFeeDecimal.toFixed(5)).toString()
-  )
+  gasParamsMVX.value.gasFee = new Decimal(
```
Comment (mmjee) [inline src/pages/SendTokens.vue]: `toDecimalPlaces` returns a `Decimal`, right? Why make a new one?

**Review by mmjee — APPROVED** [review]

**Pull request merged**